packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -0,0 +1,257 @@
+import _, { messages } from 'intl'
+import ActionButton from 'action-button'
+import decorate from 'apply-decorators'
+import Page from '../../page'
+import PropTypes from 'prop-types'
+import styles from './index.css'
+import React, { Component } from 'react'
+import Wizard, { Section } from 'wizard'
+import { map } from 'lodash'
+import { Toggle } from 'form'
+import { connectStore } from 'utils'
+import { injectIntl } from 'react-intl'
+import { linkState } from 'reaclette-utils'
+import { Container, Row, Col } from 'grid'
+import { injectState, provideState } from 'reaclette'
+import { SelectPif, SelectPool } from 'select-objects'
+import { createGetObject, getIsPoolAdmin } from 'selectors'
+import { createBondedNetwork, createNetwork, getBondModes } from 'xo'
+
+const getInitialState = () => ({
+  bonded: false,
+  bondMode: undefined,
+  description: '',
+  mtu: '',
+  name: '',
+  pif: undefined,
+  pifs: undefined,
+  vlan: '',
+})
+
+const NewNetwork = decorate([
+  connectStore(() => ({
+    isPoolAdmin: getIsPoolAdmin,
+    pool: createGetObject((_, props) => props.location.query.pool),
+  })),
+  injectIntl,
+  provideState({
+    initialState: () => ({ ...getInitialState(), bondModes: undefined }),
+    effects: {
+      linkState,
+      initialize: async () => {
+        const bondModes = await getBondModes()
+        return { bondModes }
+      },
+      onChangeBondedNetwork: () => ({ bonded, bondModes }) => ({
+        ...getInitialState(),
+        bonded: !bonded,
+        bondMode: !bonded ? bondModes[0] : undefined,
+      }),
+      onChangePif: (_, value) => ({ bonded }) =>
+        bonded ? { pifs: value } : { pif: value },
+      reset: getInitialState,
+    },
+    computed: {
+      pifPredicate: (_, { pool }) => pif =>
+        pif.vlan === -1 && pif.$host === (pool && pool.master),
+    },
+  }),
+  injectState,
+  class extends Component {
+    static contextTypes = {
+      router: PropTypes.object,
+    }
+
+    _create = () => {
+      const { pool, state } = this.props
+      const {
+        bonded,
+        bondMode,
+        description,
+        mtu,
+        name,
+        pif,
+        pifs,
+        vlan,
+      } = state
+      return bonded
+        ? createBondedNetwork({
+            bondMode,
+            description,
+            mtu,
+            name,
+            pifs: map(pifs, 'id'),
+            pool: pool.id,
+            vlan,
+          })
+        : createNetwork({
+            description,
+            mtu,
+            name,
+            pif: pif.id,
+            pool: pool.id,
+            vlan,
+          })
+    }
+
+    _selectPool = pool => {
```

Comment:
Can't this be an effect?

Reply:
`_selectPool`  use the `context` object.